src/main/kotlin/sh/hsp/techtree/FileSystemYamlReader.kt

```diff
@@ -5,15 +5,65 @@ import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
 import com.fasterxml.jackson.module.kotlin.readValue
 import com.fasterxml.jackson.module.kotlin.registerKotlinModule
 import java.io.File
+import java.io.IOException
 import java.net.URL
 
 class FileSystemYamlReader : YamlReader {
     override fun readToModel(resource: URL): TreeModel {
         val mapper = ObjectMapper(YAMLFactory()).registerKotlinModule()
         val file = File(resource.toURI())
         val yamlData = file.readText()
+        val treeModel: TreeModel = mapper.readValue(yamlData)
 
-        return mapper.readValue(yamlData)
+        validateTreeModel(treeModel)
+
+        return treeModel
+    }
+
+    private fun validateTreeModel(treeModel: TreeModel) {
+        val titleList = treeModel.nodes.drop(1).map { it.title }
```

Comment:
Nie wiem po co drop(1).
Fajny ogólnie pomysł.
Zastanawiam się tylko, czy nie musisz zrobić obu list unikalnych albo przez jakąś funkcje uniqe, czy poprzez zmapowanie do setów.

Reply:
Drop(1) był po to, aby nie liczyć head nodu. Zakładałem, że jeden TreeModel to jedno drzewo. Czyli że każdy node prowadzi do tego głównego. Dodając do osobnych list wszystkie titles i requires i usuwając tego głównego node można było porównać czy te listy są takie same. Jak się nakładały to ok, jeżeli nie to coś jest nie tak i dalej jest generacja wiadomości błędu.